### ISSUE TITLE
Apparent error in multihead attention calculation

### ISSUE THREAD

**Issue by str4w** [keras-team/keras#14947]
### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in Keras)**:

No
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Colab (and others, but can be demoed on colab)

-   **TensorFlow installed from (source or binary)**:
Colab default

-   **TensorFlow version (use command below)**:
tensorflow version 2.5.0
keras version 2.5.0

-   **Python version**:
sys.version_info(major=3, minor=7, micro=11, releaselevel='final', serial=0)
-   **Bazel version (if compiling from source)**:
NA
-   **GPU model and memory**:
CPU
-   **Exact command to reproduce**:
See this colab:  https://colab.research.google.com/drive/19hs764rkf2pgM49nNkhlw-ojzchxdhTa?usp=sharing

### Describe the problem
The multihead attention layer should multiply the k and q matrices by the value, take the dot prod, and then combine the resulting weighted sum to get the output.  However, upon testing with a random input, and freshly initialized weights (which should also be random, the output is identical at every time step, despite the attention heads having plausible, and varying values per time step.

When an attention mask is used - the values for every head (which were expected to become the reweighted first part of the sequence, with the masked part having zero weight) now become identical across all time steps.

**Comment by str4w** [comment]
The values which appear identical are not actually identical.  The differences are just extremely tiny and not visually perceptible, which was unexpected.
